extensions/wikia/CrosslinkTag/CrosslinkTagHelper.class.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * Class CrosslinkTagHelper
+ */
+class CrosslinkTagHelper extends WikiaModel {
+
+	const CACHE_TTL = 3600;
+	const FANDOM_API_URL = 'http://fandom.wikia.com/wp-json/wp/v2/';
+	const VALID_HOST = 'fandom.wikia.com';
+
+	protected $endpoints = [
+		'articles' => 'posts',
+		'videos' => 'video',
+	];
+
+	/**
+	 * Show the crosslink unit only on Article pages and Main pages
+	 * @return boolean
+	 */
+	public function canShowUnit() {
+		if ( !$this->app->checkSkin( 'oasis' ) ) {
+			return false;
+		}
+
+		$title = $this->wg->Title;
+		if ( $title instanceof Title && $title->getNamespace() == NS_MAIN ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get data for the article by slug
+	 * @param string $slug - alphanumeric id
+	 * @param string $pageType - page type
+	 * @return array|false
+	 */
+	public function getArticleDataBySlug( $slug, $pageType = 'articles' ) {
+		$cacheKey = $this->getMemcKey( $slug );
+		$data = $this->wg->Memc->get( $cacheKey );
+		if ( !is_array( $data ) ) {
+			$params = [
+				'_embed' => 1,
+				'slug' => $slug,
+			];
+
+			if ( array_key_exists( $pageType, $this->endpoints ) ) {
+				$endpoint = $this->endpoints[$pageType];
+			} else {
+				$endpoint = $this->endpoints['articles'];
+			}
+
+			$apiUrl = self::FANDOM_API_URL . $endpoint.'?' . http_build_query( $params );
+			$method = 'GET';
+			$options = [ 'noProxy' => true ];
```

Comment:
I think we should add some timeout in here. It's happening in the parser tag after all